### ISSUE TITLE
When is 0.84.4 (2022.10.0) coming?

### ISSUE THREAD

**Issue by BridgeHeadland** [joncampbell123/dosbox-x#3774]
### Question

Usually it's exactly one month, or almost, before the next official version of DOSBox-X becomes available for download, but I haven't seen anything for the October 2022 version so far. I can't find any information about a next version anywhere either.

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
The release is held up for two reasons: My professional work is gearing up for code freeze and a release, and there is an issue on Mac OS in which the IME falsely translates all spaces to doublewide spaces in PC-98 mode which makes normal DOS prompt use impossible. I'll fix that Mac OS X issue and then issue a release.

**Comment by joncampbell123** [comment]
`Add the following functions.
bool IME_GetEnable()
{
  TISInputSourceRef is = TISCopyCurrentKeyboardInputSource();
  CFBooleanRef ret = (CFBooleanRef)TISGetInputSourceProperty(is, kTISPropertyInputSourceIsASCIICapable);
  return !CFBooleanGetValue(ret);
}
Modified the processing of case SDL_KEYDOWN: in sdlmain.cpp GFX_Events() as follows.
  if(dos.im_enable_flag) {
#if defined (WIN32) && !defined(HX_DOS) || defined(MACOSX) && defined(SDL_DOSBOX_X_IME)
    if(event.type == SDL_KEYDOWN && IME_GetEnable()) {
      // Enter, BS, TAB, <-, ->

I don't have a mac that I can test this on, so please give it a try.`

**Comment by joncampbell123** [comment]
"I found the following by doing various searches in Japanese using "mac IME" etc.
https://github.com/minoki/InputSourceSelector
I searched for TISGetInputSourceProperty in the source and found the following
https://gist.github.com/kawaz/3643468aab25c7a8c7bcf80db9cdb10e
Looking at the result display, it seems that kTISPropertyInputSourceIsASCIICapable can be used to determine the IME status.
https://egg.5ch.net/test/read.cgi/mac/1414566499/33
I quoted the above source that I found by searching further, but it says it can detect only the first time (in Japanese), so you need to confirm that it works on current OS.
I hope the OS side behavior is fixed now since it was written in 2014."

**Comment by maron2000** [comment]
I tested commit 211b4dd97d1aa92099967d4a531722fe36141818 on Intel Mac (Monterey 12.6).
I can type commands.
Typing Japanese characters is kind of glitchy, but not impossible.
If you can tell me what to test, maybe I can try.

![mac_pc98test](https://user-images.githubusercontent.com/68574602/195492167-ecd6e1be-6e5d-469f-a928-c35490d9cc48.png)

**Comment by maron2000** [comment]
I tried commit 211b4dd97d1aa92099967d4a531722fe36141818 on Intel Mac Monterey(12.6)
I can type command and input Japanese characters.
If you can tell me what else to try, maybe I can test.
![mac_pc98test](https://user-images.githubusercontent.com/68574602/195492902-aee2bb27-3995-4d3a-87d8-e8ed844663c3.png)

**Comment by nanshiki** [comment]
Thanks for the confirmation.
The #if defined (MACOSX) block in sdmain.cpp should be fine to remove.
Currently, a build error occurs on Windows. For Windows, && after IME_GetEnable() is unnecessary.
```
   if(dos.im_enable_flag) {
#if defined (WIN32) && !defined(HX_DOS) || defined(MACOSX) && defined(SDL_DOSBOX_X_IME)
       if(event.type == SDL_KEYDOWN && IME_GetEnable()) {
           // Enter, BS, TAB, <-, ->
           if(event.key.keysym.sym == 0x0d || event.key.keysym.sym == 0x08 || event.key.keysym.sym == 0x09 || event.key.keysym.scancode == 0x4f || event.key.keysym.scancode == 0x50) {
```

I think that the composition string is not displayed in the case of macOS.
It seems necessary to draw the aString passed by setMarkedText on your own.

**Comment by maron2000** [comment]
@nanshiki @joncampbell123 
The fix seems to work fine, and build was successful for all platforms.
Tested x64 VS build and Mac.
I will put a PR on behalf of @nanshiki .

![ime_test_macos](https://user-images.githubusercontent.com/68574602/195526770-eae9912d-5462-4446-8243-94c22a705e7e.png)
![ime_test_vsbuild](https://user-images.githubusercontent.com/68574602/195526788-aaf7f661-2e0f-4e87-851f-5aa0893c5bf8.png)

**Comment by nanshiki** [comment]
@maron2000 Thanks.
I think there are still problems, but I'm glad that it seems to work as expected.

**Comment by maron2000** [comment]
@nanshiki 
Indeed, I found that if you type Hiragana with IME, it doesn't show up inline.
But at least you can manage to type both commands and Japanese, I think we can go ahead to release the next version.
